### ISSUE TITLE
json_extract consumes too much memory

### ISSUE THREAD

**Issue by izellevy** [duckdb/duckdb#8734]
### What happens?

Reading the same properties using json_extract allocates the memory only once however if I try to read different properties I see that duckdb memory consumption significantly increases. 

For example:

Let's say running this query consumes X MB memory. 

`select json_extract(data, "some_field") from table`

I would expect this select to consume relatively similar memory but it consumes 2X.

`select json_extract(data, "some_field"), json_extract(data, "some_other_field") from table`

If I have 100 columns of json_extract then in the end we end up with 100X memory consumption.

### To Reproduce

Install pytest-memray save the following code to a file and run pytest --memray your_file.py. This consumes around 12.7GiB memory for relatively small datasets. You can play with the COL_SIZE to see that memory linearly increases.

Output:

     📦 Total memory allocated: 12.7GiB
         📏 Total allocations: 868141
         📊 Histogram of allocation sizes: |  █      |
         🥇 Biggest allocating functions:
                - test_memory:/Users/.../Library/Application Support/JetBrains/PyCharm2023.2/scratches/emlite/duckdb_perf.py:22 -> 12.5GiB
                - test_memory:/Users/.../Library/Application Support/JetBrains/PyCharm2023.2/scratches/emlite/duckdb_perf.py:17 -> 64.0MiB
                - iterencode:/opt/homebrew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py:257 -> 20.7MiB
                - _compile_bytecode:<frozen importlib._bootstrap_external>:640 -> 2.5MiB

```python

import json
import duckdb
import pyarrow as pa

ROW_SIZE = 1000
COL_SIZE = 100
STRING_SIZE = 200


def test_memory():
    with duckdb.connect() as con:
        my_value = "a" * STRING_SIZE
        row = {f"col_{idx}": f"{my_value}_{idx}" for idx in range(COL_SIZE)}

        rows = [json.dumps(row) for _ in range(ROW_SIZE)]
        con.register("my_view", pa.Table.from_pydict({"data": pa.array(rows)}))
        con.execute("create table my_table (data JSON)")
        con.execute(f"insert into my_table (select data from my_view)")

        all_extracts = ",".join([f"json_extract(data, '{key}')" for key in row.keys()])
        con.execute(f"select {all_extracts} from my_table").fetchall()
```

### OS:

MacOS 13.2.1 aarch64

### DuckDB Version:

0.8.1

### DuckDB Client:

Python

### Full Name:

Izel Levy

### Affiliation:

Explorium

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by lnkuiper** [comment]
You can currently get around this by using the following syntax:
```sql
select json_extract(data, ['some_field', 'some_other_field'])
from table
```
Which will return a list of the extracted values. If you want them as columns, you can do:
```sql
with extracted as (
  select json_extract(data, ['some_field', 'some_other_field']) my_list
  from table
)
select my_list[1], my_list[2]
from extracted
```

Of course, DuckDB should recognize this, optimize the query, and do this for you, but that is not a simple feature, and is left for future work.

**Comment by izellevy** [comment]
Hey @lnkuiper thanks for the response! Maybe it is worth adding a warning to the docs. It costed us a decent amount of time to debug and understand this issue.
